sockeye/beam_search.py

```diff
@@ -495,9 +523,13 @@ def forward(self, best_hyp_indices, *states):
 def _get_vocab_slice_ids(restrict_lexicon: Optional[lexicon.TopKLexicon],
                          source_words: pt.Tensor,
                          eos_id: int,
-                         beam_size: int) -> Tuple[pt.Tensor, int]:
+                         beam_size: int,
+                         target_prefix: Optional[pt.Tensor] = None) -> Tuple[pt.Tensor, int]:
     device = source_words.device
     vocab_slice_ids = restrict_lexicon.get_trg_ids(source_words.cpu().int().numpy())
+    if target_prefix is not None:
+        # Ensuring that target prefix ids are part of vocab_slice_ids
+        vocab_slice_ids = onp.lib.arraysetops.unique(onp.concatenate([vocab_slice_ids, target_prefix.cpu().int().numpy().flatten()]))
```

Comment:
Alternatively, you could first convert `vocab_slice_ids` to pytorch and use pytorch's unique.

Reply:
Thanks Tobias for the suggestion. I tried and noticed a nice improvement in speed (from 6.0s to 5.4s in my test, around 10% improvement) by converting vocab_slice_ids to pytorch and use pytorch's unique function. Yeah, so let's go for it.

Testing code for speed:
```
import timeit
set_up_line = """
import numpy as np
import torch as pt
pt.manual_seed(1)
np.random.seed(1)
vocab_slice_idss = np.random.randint(0, 10, (100, 50))
target_prefix = pt.randint(0, 10, (100, 50))
"""
executive_line = """
z = target_prefix.device
for i in range(target_prefix.size(0)):
    vocab_slice_ids = np.lib.arraysetops.unique(np.concatenate([vocab_slice_idss[i], target_prefix[i].cpu().int().numpy().flatten()]))
    vocab_slice_ids = pt.nn.functional.pad(pt.tensor(vocab_slice_ids, dtype=pt.int64, device=z),  # type: ignore
                                           pad=(0, 7 - ((vocab_slice_ids.size - 1) % 8)),
                                           mode='constant', value=2)
"""
timeit.timeit(executive_line,setup=set_up_line, number=1000)
6.065197644056752


executive_line = """
z = target_prefix.device
for i in range(target_prefix.size(0)):
	vocab_slice_ids = pt.tensor(vocab_slice_idss[i])
	vocab_slice_ids = pt.unique(pt.concat([vocab_slice_ids.type(pt.int64).to(z), target_prefix[0].type(pt.int64).to(z)], -1))
	vocab_slice_ids = pt.nn.functional.pad(vocab_slice_ids.clone().detach(),  # type: ignore
	                                           pad=(0, 7 - ((vocab_slice_ids.size(-1) - 1) % 8)),
	                                           mode='constant', value=2)
                                           """
timeit.timeit(executive_line,setup=set_up_line, number=1000)
5.411461449926719
```

Reply:
Great, thanks for benchmarking!